src/l2_tracker/events.h

```diff
@@ -72,6 +74,63 @@ struct NewServiceNode : L2StateChange {
     static constexpr std::string_view description = "new SN"sv;
 };
 
+struct ContributorV2 {
+    enum class Version {
+        version_invalid,
+        version_0,
+        _count,
+    };
+
+    eth::address address;
+    eth::address beneficiary;
+    uint64_t     amount;
+
+    auto operator<=>(const ContributorV2& o) const = default;
+
+    template <class Archive>
+    void serialize_object(Archive& ar) {
+        auto version = tools::enum_top<Version>;
+        field_varint(ar, "version", version, [](auto v) { return v < Version::_count; });
```

Comment:
This is allowing 0 == `version_invalid` as a valid de-serializable value.

Reply:
Fixed